static/js/publisher/release/reducers/pendingReleases.js

```diff
@@ -2,39 +2,44 @@ import {
   RELEASE_REVISION,
   UNDO_RELEASE,
   CANCEL_PENDING_RELEASES,
-  SET_PROGRESSIVE_RELEASE_PERCENTAGE
+  SET_PROGRESSIVE_RELEASE_PERCENTAGE,
+  UPDATE_PROGRESSIVE_RELEASE_PERCENTAGE,
+  PAUSE_PROGRESSIVE_RELEASE,
+  RESUME_PROGRESSIVE_RELEASE
 } from "../actions/pendingReleases";
+
 import { CLOSE_CHANNEL } from "../actions/pendingCloses";
 
 function removePendingRelease(state, revision, channel) {
-  if (state[revision.revision]) {
-    const channels = [...state[revision.revision].channels];
+  const releaseKey = `${revision.revision}-${channel}`;
```

Comment:
Why changing the key to be per revision-channel rather than just revision?

Reply:
We were adding to the channel array if you were releasing the same revision to 2 or more channels.
This meant there was only 1 progressive release object for both of those channels.
This in turn meant you could not independently update a progressive release on one channel, and add a new progressive release on another channel with a different key/ percentage.

I thought about changing the progressive release object to be channel keyed:
`progressive: { ${channel}: {}}` but it became complicated very fast - this seemed simpler. Open to suggestions though!

Reply:
Ah, good point. 

Previous pending releases structure was 1:1 reflecting what we send to API, each 'key' (revision) was single API call (releasing a revision to list of channels).
It would be a pity if we loose that - if there is a key per revision-channel now, does it mean even standard "100%" releases are separated when on different channels? Releasing a revision to multiple channels would mean multiple API calls now?

I also need to check if there is no other code that relied on the fact that prendingRelease key is a revision (like counting how many revisions are to release or something)...

Reply:
That's what I was afraid:

Revisions to release are not counted properly now (when single revision is released to multiple channels):
<img width="602" alt="Screenshot 2019-11-05 at 10 57 11" src="https://user-images.githubusercontent.com/83575/68197888-531eb780-ffbb-11e9-9498-c5bf367aeaab.png">


And the releases via API now happen in multiple calls (one per channel) instead of just one:

<img width="505" alt="Screenshot 2019-11-05 at 10 57 41" src="https://user-images.githubusercontent.com/83575/68197877-4dc16d00-ffbb-11e9-9bad-14ac1cd5b2c0.png">


Changes are certainly needed to the model, but we shouldn't make most common and default experience worse by adding progressive releases.